linux/opt/sage/bin/make_env_vars_file.sh

```diff
@@ -1,27 +1,65 @@
 #! /bin/bash
 
+# Creates a file which sets environment variables. These are common variables
+# used by other init scripts that set up an AWS EC2 Linux instance.
+
+set -ex
+
+usage() {
+  msg="To run create_env_vars_file.sh, ensure that AWS_REGION, STACK_ID, "
+  msg+="and STACK_NAME are set and not empty."
+  echo "$msg"
+}
+
+if [[ -z "$AWS_REGION" || -z "$STACK_ID" || -z "$STACK_NAME" ]]; then
+  usage >&2
+  exit 1
+fi
+
 EC2_INSTANCE_ID=$(/usr/bin/curl -s http://169.254.169.254/latest/meta-data/instance-id)
 ROOT_DISK_ID=$(/usr/bin/aws ec2 describe-volumes \
-  --region $AWS_REGION \
-  --filters Name=attachment.instance-id,Values=$EC2_INSTANCE_ID \
+  --region "$AWS_REGION" \
+  --filters Name=attachment.instance-id,Values="$EC2_INSTANCE_ID" \
   --query Volumes[].VolumeId --out text)
-EC2_INSTANCE_TAGS=$(aws --region $AWS_REGION \
+EC2_INSTANCE_TAGS=$(aws --region "$AWS_REGION" \
   ec2 describe-tags \
-  --filters Name=resource-id,Values=$EC2_INSTANCE_ID)
+  --filters Name=resource-id,Values="$EC2_INSTANCE_ID")
+
 extract_tag_value() {
-  echo $EC2_INSTANCE_TAGS | jq -j --arg KEYNAME "$1" '.Tags[] | select(.Key == $KEYNAME).Value '
+  echo "$EC2_INSTANCE_TAGS" | jq -j --arg KEYNAME "$1" '.Tags[] | select(.Key == $KEYNAME).Value '
 }
+
 DEPARTMENT=$(extract_tag_value Department)
 PROJECT=$(extract_tag_value Project)
 PROVISIONING_PRINCIPAL_ARN=$(extract_tag_value 'aws:servicecatalog:provisioningPrincipalArn')
-OWNER_EMAIL=${!PROVISIONING_PRINCIPAL_ARN##*/}
-
-echo "export AWS_REGION=$AWS_REGION" > /opt/sage/bin/instance_env_vars.sh
-echo "export STACK_NAME=$STACK_NAME" >> /opt/sage/bin/instance_env_vars.sh
-echo "export STACK_ID=$STACK_ID" >> /opt/sage/bin/instance_env_vars.sh
-echo "export EC2_INSTANCE_ID=$EC2_INSTANCE_ID" >> /opt/sage/bin/instance_env_vars.sh
-echo "export ROOT_DISK_ID='$ROOT_DISK_ID'" >> /opt/sage/bin/instance_env_vars.sh
-echo "export DEPARTMENT=$DEPARTMENT" >> /opt/sage/bin/instance_env_vars.sh
-echo "export PROJECT=$PROJECT" >> /opt/sage/bin/instance_env_vars.sh
-echo "export OWNER_EMAIL=$OWNER_EMAIL" >> /opt/sage/bin/instance_env_vars.sh
-chmod +x /opt/sage/bin/instance_env_vars.sh
+PRINCIPAL_ID=${!PROVISIONING_PRINCIPAL_ARN##*/}
+
+if [[ "$PRINCIPAL_ID" =~ [[:digit:]] ]]; then
+  USER_PROFILE_RESPONSE=$(curl -s "https://repo-prod.prod.sagebase.org/repo/v1/userProfile/$PRINCIPAL_ID")
+  SYNAPSE_USERNAME=$(echo "$USER_PROFILE_RESPONSE" | jq '.userName')
+  if [[ -z "$SYNAPSE_USERNAME" || "$SYNAPSE_USERNAME" == "null" ]]; then
+    echo "Could not extract user name from Synapse user profile response: $USER_PROFILE_RESPONSE" >&2
+    exit 1
+  fi
+  OWNER_EMAIL="$SYNAPSE_USERNAME@synapse.org"
+elif [[ "$PRINCIPAL_ID" =~ .*"@".* ]]; then
```

Comment:
Deliberately keeping this regex simple.

Reply:
Thanks for sharing. Wherever this regex was first added, it looks like it was not in this file. I've used `git blame` on a few and haven't found the original commit.
Also, this is java-flavored regex. I need ERE.

Further -- while it might be nice to have a good email regex, I am making the case that this is overbuilding. Our `PRINCIPAL_ID` comes from two possible sources:
1. IAM accounts where we use sagebase email addresses
2. Syn login where the value will be all digits.

@zaro0508 what do you think? If you disagree, are you good at converting complicated REGEX from one format to another? Because I know this would be a big time sink for me.